corehq/apps/userreports/reports/view.py

```diff
@@ -281,6 +282,117 @@ def lang(self):
+
+    @property
+    @memoized
+    def async_response(self):
+        """
+        Renders the asynchronous view of the report template, returned as json.
+        """
+        return JsonResponse(self._async_context())
+
+    def _async_context(self):
+        """
+        Prepares the context for the asynchronous request.
+        """
+        self.update_template_context()
+        self.update_report_context()
+
+        rendered_filters = None
+        # Check if 'hq_filters' parameter is present
+        if self.request.GET.get('hq_filters', False):
+            self.update_filter_context()
+
+            rendered_filters = render_to_string(
+                self.template_filters, context=self.main_context, request=self.request
+            )
+
+        # If we're not just requesting filters, prepare the report data
+        if not self.request.GET.get('hq_filters', False):
+            try:
+                # Fetch data for the report
+                if hasattr(self, 'data_source') and self.data_source:
+                    self.data_source.set_filter_values(self.filter_values)
+                    data = list(self.data_source.get_data())
+                    total_row = self.data_source.get_total_row() if self.data_source.has_total_row else None
+
+                    # Update the context with the fetched data
+                    self.data_source.data = data
+                    self.data_source.total_row = total_row
+            except Exception:
+                # Continue to render the template even if there's an error
+                pass
+
+        rendered_report = render_to_string(
+            self.template_report, self.main_context, request=self.request
+        )
+
+        report_table_js_options = self.main_context.get('report_table_js_options', {})
+
+        return {
+            'filters': rendered_filters,
+            'report': rendered_report,
+            'report_table_js_options': report_table_js_options,
+            'title': self.spec.title,
+            'slug': self.slug,
+            'url_root': self.url,
+        }
+
+    def update_template_context(self):
+        """
+        Update the template context with common values.
+        """
+        self.main_context.update({
+            'report': {
+                'title': self.spec.title,
+                'description': self.spec.description,
+                'section_name': self.section_name,
+                'spec': self.spec,
+                'status': self.spec.status if hasattr(self.spec, 'status') else None,
+                'slug': self.slug,
+            }
+        })
+
+    def update_report_context(self):
+        """
+        Update the context for the report content.
+        """
+        if hasattr(self, 'data_source') and self.data_source:
+            report_config = self.spec
+
+            # Initialize data and total_row attributes
+            self.data_source.data = []
+            self.data_source.total_row = None
```

Comment:
I am bit hesitant about adding attributes on the fly that are not originally not present or initialized by the object's class itself.
Can we have these as separate in context itself? `data_source_data` and `data_source_total_row`?